[Flake] Integration test flake - TestServiceDiscoverDR

```
=== RUN   TestServiceDiscoverDR
{"level":"error","time":"2024-02-19T14:25:49Z","message":"Err: 'exit status 1'. stderr: 'Error from server (InternalError): error when creating \"/home/runner/work/kiali/kiali/tests/integration/assets/bookinfo-destination-rule-reviews.yaml\": Internal error occurred: failed calling webhook \"validation.istio.io\": failed to call webhook: Post \"[https://istiod.istio-system.svc:443/validate?timeout=10s\](https://istiod.istio-system.svc/validate?timeout=10s\)": dial tcp 10.96.189.243:443: connect: connection refused\n'."}
    services_test.go:135: 
        	Error Trace:	/home/runner/work/kiali/kiali/tests/integration/tests/services_test.go:135
        	Error:      	Should be true
        	Test:       	TestServiceDiscoverDR
{"level":"error","time":"2024-02-19T14:25:49Z","message":"exit status 1"}
--- FAIL: TestServiceDiscoverDR (0.65s)
FAIL
exit status 1
FAIL	github.com/kiali/kiali/tests/integration/tests	364.944s
Error: Process completed with exit code 1.
```

--- a/tests/integration/tests/remote_istiod_test.go
+++ b/tests/integration/tests/remote_istiod_test.go
@@ -70,6 +70,8 @@ func TestRemoteIstiod(t *testing.T) {
 		}
 		_, err = kubeClient.AppsV1().Deployments(originalConf.IstioNamespace).Update(ctx, istiod, metav1.UpdateOptions{})
 		require.NoError(err)
+
+		require.NoError(kube.WaitForDeploymentReady(ctx, kubeClient, originalConf.IstioNamespace, istioDeploymentName))
 	})
 
 	// Expose the istiod /debug endpoints by adding a proxy to the pod.

--- a/tests/integration/utils/kiali/instance.go
+++ b/tests/integration/utils/kiali/instance.go
@@ -18,6 +18,7 @@ import (
 
 	"github.com/kiali/kiali/config"
 	"github.com/kiali/kiali/log"
+	"github.com/kiali/kiali/tests/integration/utils/kube"
 )
 
 var kialiGroupVersionResource = schema.GroupVersionResource{Group: "kiali.io", Version: "v1alpha1", Resource: "kialis"}
@@ -218,40 +219,6 @@ func restartDeployment(ctx context.Context, clientset kubernetes.Interface, name
 	return retryErr
 }
 
-func waitForDeploymentReady(ctx context.Context, clientset kubernetes.Interface, namespace, deploymentName string) error {
-	timeout := 5 * time.Minute
-	pollInterval := 10 * time.Second
-
-	return wait.PollUntilContextTimeout(ctx, pollInterval, timeout, true, func(ctx context.Context) (bool, error) {
-		deployment, err := clientset.AppsV1().Deployments(namespace).Get(ctx, deploymentName, metav1.GetOptions{})
-		if err != nil {
-			return false, err
-		}
-
-		if deployment.Generation != deployment.Status.ObservedGeneration {
-			log.Debug("The deployment has not observed the latest spec update yet.")
-			return false, nil
-		}
-
-		if deployment.Status.Replicas != *deployment.Spec.Replicas {
-			log.Debugf("Waiting for deployment to be fully rolled out (%d/%d replicas)", deployment.Status.Replicas, *deployment.Spec.Replicas)
-			return false, nil
-		}
-
-		if deployment.Status.UpdatedReplicas != *deployment.Spec.Replicas {
-			log.Debugf("Waiting for deployment to be updated (%d/%d replicas)", deployment.Status.UpdatedReplicas, *deployment.Spec.Replicas)
-			return false, nil
-		}
-
-		if deployment.Status.ReadyReplicas != *deployment.Spec.Replicas {
-			log.Debugf("Waiting for deployment to be ready (%d/%d replicas)", deployment.Status.ReadyReplicas, *deployment.Spec.Replicas)
-			return false, nil
-		}
-
-		return true, nil
-	})
-}
-
 // Restart will recreate the Kiali pod and wait for it to be ready.
 func (in *Instance) Restart(ctx context.Context) error {
 	log.Debug("Restarting Kiali deployment")
@@ -260,7 +227,7 @@ func (in *Instance) Restart(ctx context.Context) error {
 	}
 
 	log.Debug("Waiting for Kiali deployment to be ready")
-	if err := waitForDeploymentReady(ctx, in.kubeClient, in.ResourceNamespace, in.Name); err != nil {
+	if err := kube.WaitForDeploymentReady(ctx, in.kubeClient, in.ResourceNamespace, in.Name); err != nil {
 		return err
 	}
 

--- a/tests/integration/utils/kube/util.go
+++ b/tests/integration/utils/kube/util.go
@@ -1,0 +1,47 @@
+package kube
+
+import (
+	"context"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/kiali/kiali/log"
+)
+
+// WaitForDeploymentReady waits for a deployment to be fully rolled out and ready.
+func WaitForDeploymentReady(ctx context.Context, clientset kubernetes.Interface, namespace, deploymentName string) error {
+	timeout := 5 * time.Minute
+	pollInterval := 10 * time.Second
+
+	return wait.PollUntilContextTimeout(ctx, pollInterval, timeout, true, func(ctx context.Context) (bool, error) {
+		deployment, err := clientset.AppsV1().Deployments(namespace).Get(ctx, deploymentName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		if deployment.Generation != deployment.Status.ObservedGeneration {
+			log.Debug("The deployment has not observed the latest spec update yet.")
+			return false, nil
+		}
+
+		if deployment.Status.Replicas != *deployment.Spec.Replicas {
+			log.Debugf("Waiting for deployment to be fully rolled out (%d/%d replicas)", deployment.Status.Replicas, *deployment.Spec.Replicas)
+			return false, nil
+		}
+
+		if deployment.Status.UpdatedReplicas != *deployment.Spec.Replicas {
+			log.Debugf("Waiting for deployment to be updated (%d/%d replicas)", deployment.Status.UpdatedReplicas, *deployment.Spec.Replicas)
+			return false, nil
+		}
+
+		if deployment.Status.ReadyReplicas != *deployment.Spec.Replicas {
+			log.Debugf("Waiting for deployment to be ready (%d/%d replicas)", deployment.Status.ReadyReplicas, *deployment.Spec.Replicas)
+			return false, nil
+		}
+
+		return true, nil
+	})
+}

